### PR TITLE
Removing DividerCommons type in @fluentui/react-divider

### PR DESCRIPTION
**Pull request by khmakoto**
## PR Description

This PR removes the `Commons` type pattern in the types for our `Divider` component.

## Related Issue(s)

Fixes part of #22862

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                             |    Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                Change |
| :-------------------------------------------------------------------------------------------- | --------------------------: | --------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr> | `15.484 kB`<br />`5.581 kB` | `15.504 kB`<br />`5.587 kB` | `20 B` <br />`6 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `179.701 kB`<br />`50.147 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.066 kB`<br />`11.089 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d1112b254407f0f0c27119326ec86d2112bc6d70'>d1112b254407f0f0c27119326ec86d2112bc6d70</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/Avatar-mount_base.html">827</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/Avatar-mount.html">813</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/Button-mount_base.html">533</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/Button-mount.html">492</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/FluentProvider-mount_base.html">1809</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/FluentProvider-mount.html">1844</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/FluentProviderWithTheme-mount_base.html">283</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/FluentProviderWithTheme-mount.html">264</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">204</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">233</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">287</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">315</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/MakeStyles-mount_base.html">1499</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22951/perf-test-react-components/MakeStyles-mount.html">1572</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-divider/src/components/Divider/useDividerStyles.ts [hunk 1]
```diff
@@ -33,20 +33,16 @@ const useBaseStyles = makeStyles({
     lineHeight: tokens.lineHeightBase200,
     textAlign: 'center',
 
-    color: tokens.colorNeutralForeground2,
```
Comment (spmonahan) [inline packages/react-divider/src/components/Divider/useDividerStyles.ts:36]: It looks like the `subtle` and `strong` appearances were relying on this as the color being set here as they don't have overrides for `color`. I think they need to have this color added to their styles.

File: packages/react-divider/src/components/Divider/Divider.types.ts [hunk 2]
```diff
@@ -12,32 +12,36 @@ export type DividerSlots = {
...
    */
-  vertical: boolean;
+  vertical?: boolean;
 };
 
-export type DividerProps = ComponentProps<Partial<DividerSlots>> & Partial<DividerCommons>;
-
-export type DividerState = ComponentState<DividerSlots> & DividerCommons;
+export type DividerState = ComponentState<DividerSlots> &
+  Required<Pick<DividerProps, 'alignContent' | 'appearance' | 'inset' | 'vertical'>>;
```
Comment (spmonahan) [inline packages/react-divider/src/components/Divider/Divider.types.ts:47]: Is `Pick` being used here to be explicit? Since all the keys are being picked it isn't really necessary.

Comment (khmakoto) [reply packages/react-divider/src/components/Divider/Divider.types.ts:47]: This is done so we don't get all the HTML props coming from `ComponentProps<DividerSlots>` in `ComponentState`.

File: packages/react-divider/src/components/Divider/useDividerStyles.ts (hunk 1 above)
Comment (khmakoto) [reply packages/react-divider/src/components/Divider/useDividerStyles.ts:36]: Done!

**Review by spmonahan — APPROVED** [review]

File: packages/react-divider/src/components/Divider/useDividerStyles.ts [hunk 3]
```diff
@@ -95,7 +91,20 @@ const useBaseStyles = makeStyles({
       ...shorthands.borderColor(tokens.colorBrandStroke1),
     },
   },
+  default: {
+    color: tokens.colorNeutralForeground2,
```
Comment (GeoffCoxMSFT) [inline packages/react-divider/src/components/Divider/useDividerStyles.ts:95]: Nit: You could break out the style changes to another PR to keep this one focused on just the commons change.

**Review by GeoffCoxMSFT — APPROVED** [review]

File: packages/react-divider/src/components/Divider/useDividerStyles.ts (hunk 3 above)
Comment (khmakoto) [reply packages/react-divider/src/components/Divider/useDividerStyles.ts:95]: I unfortunately cannot do that because this change also makes some of the commons props required in state (which they should have been all along tbh) and to keep the `styles[state.appearance]` work I need to make these style changes.
